engine/runtime/src/test/scala/org/enso/compiler/test/semantic/TypeSignaturesTest.scala

```diff
@@ -244,6 +247,11 @@ class TypeSignaturesTest
     "resolve execution contexts" in {
       val code =
         """
+          |type A
+          |type B
+          |type C
+          |type Input
+          |
           |foo : A -> B -> C in Input
```

Comment:
Shouldn't we also have a test that makes sure the errors are properly reported, when the `A`, `B`, `C` or `Input` isn't defined? Such a _falsification_ test is needed to verify compiler detects when something is wrong. Apologies, if I missed the test, but I don't seem to recognize it in this PR.